src/bms/lex/token.rs

```diff
@@ -92,7 +92,24 @@ pub enum Token<'a> {
     /// `#EXRANK[01-ZZ] [0-3]`. Defines the judgement level change object.
     ExRank(ObjId, JudgeLevel),
     /// `#EXWAV[01-ZZ] [parameter order] [pan or volume or frequency; 1-3] [filename]`. Defines the key sound object with the effect of pan, volume and frequency.
-    ExWav(ObjId, [&'a str; 4], &'a Path),
+    ExWav {
+        /// The id of the object to define.
+        id: ObjId,
+        /// The pan of the sound. Also called volume balance.
+        /// Range: [-10000, 10000]. -10000 is leftmost, 10000 is rightmost.
+        /// Default: 0.
+        pan: ExWavPan,
+        /// The volume of the sound.
+        /// Range: [-10000, 0]. -10000 is 0%, 0 is 100%.
+        /// Default: 0.
+        volume: ExWavVolume,
+        /// The frequency of the sound. Unit: Hz.
+        /// Range: [100, 100000].
+        /// Default: None.
+        frequency: Option<ExWavFrequency>,
```

Comment:
```suggestion
        /// The pan decay of the sound. Also called volume balance.
        pan: ExWavPan,
        /// The volume decay of the sound.
        volume: ExWavVolume,
        /// The pitch frequency of the sound.
        frequency: Option<ExWavFrequency>,
```
You can remove comments about constraints. Because they are responsible for each type.